chore: 1031 migrate docs to nuxt UI docs template (#1049)
Tresjs/tres: docs/content/3.api/2.composables/1.use-tres.md

---
title: useTres
description: useTres provides a convenient access to a simplified TresJS context.
---

The `useTres` composable provides convenient access to a simplified TresJS context with direct access to core properties like the scene, renderer, camera, and utility functions. It's designed to be more straightforward than `useTresContext` while still providing access to essential TresJS functionality.

## Usage

```ts
import { useTres } from '@tresjs/core'

const { scene, renderer, camera, sizes, invalidate, advance } = useTres()

// Access the active camera
console.log('Current camera:', camera.value)

// Get canvas dimensions
console.log('Canvas size:', sizes.width.value, sizes.height.value)

// Trigger a re-render in on-demand mode
invalidate()
```

> ::warning

Reviewer: **suggestion:** The page about useTresContext has this warning before the code sample. For uniformity, I suggest moving this too.

> `useTres` can only be used inside of a `TresCanvas` component since it relies on the TresJS context provider.

Reviewer: ```suggestion
`useTres` can only be used in child components of a [`TresCanvas`](/api/components/tres-canvas) component, as its data is provided by [`TresCanvas`](/api/components/tres-canvas).
```

console.log('Canvas size:', sizes.width.value, sizes.height.value)

> // Trigger a re-render in on-demand mode

Author: @Tinoooo I don't remember how it was before.

> // Trigger a re-render in on-demand mode

Author: Added back you can check now @Tinoooo
